test/extended/openstack/utils.go

```diff
@@ -0,0 +1,119 @@
+package openstack
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	testutils "k8s.io/kubernetes/test/utils"
+
+	o "github.com/onsi/gomega"
+)
+
+func ElementExists(s []string, str string) bool {
+	for _, v := range s {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
+func RandomSuffix() string {
+	return strconv.Itoa(rand.Intn(10000))
+}
+
+func SubnetExists(networkClient *gophercloud.ServiceClient, listOpts subnets.ListOpts, ns *v1.Namespace) bool {
+	subnet_name := ns.Name
+	subnet_alt_name := "ns/" + ns.Name + "-subnet"
+	subnets := ListSubnets(networkClient, listOpts)
+	subnets_names := []string{}
+	for _, s := range subnets {
+		subnets_names = append(subnets_names, s.Name)
+	}
+	return ElementExists(subnets_names, subnet_name) || ElementExists(subnets_names, subnet_alt_name)
```

Comment:
@gryf, can you double-check this will be okay now?

Reply:
Given, that code with name changing is already merged, that's all true… until [this change ](https://review.opendev.org/c/openstack/kuryr-kubernetes/+/841105) would be accepted. than it will be additional one **or** line 35 would be amended to reflect the new name - but that depends if mentioned patch would be accepted before FF.